numba/tests/test_function.py

```diff
@@ -0,0 +1,575 @@
+import types as pytypes
```

Comment:
This file could do with a more descriptive name, perhaps `test_first_class_functions.py` or `test_function_type.py`, it would also benefit from a docstring explaining what it is testing.

Reply:
Moved to `test_function_type.py`.